### DISCUSSION TITLE
STM32F070F6P6 and hardware requirements

### DISCUSSION THREAD

**Discussion by H4w4k4** [ARMmbed/DAPLink#1075] [Q&A] [answered]
Hello,

I'm new to DAPLink. I would like to incorpore the HIC in my next project with a LPC55S69. I'm trying to find the cheapest single quantity MCU possible and I found the STM32F070F6P6. Would it work as a HIC? 

Also what are the minimal hardware requirement to run DAPLink? Are they the same as described for CMSIS-DAP [here](https://arm-software.github.io/CMSIS_5/DAP/html/group__DAP__Config__gr.html)?

**Comment by mbrossard** [comment, marked as answer]
Hi,

Compared to CMSIS-DAP, DAPLink needs more RAM and Flash for additional features (drag-n-drop flashing, WebUSB, CDC-ACM)
You can look at the list of supported HICs [https://github.com/ARMmbed/DAPLink/tree/develop/docs/hic]. The STM32F072C8T6 is probably the cheapest that would work out-of-the-box.
In general the requirements for DAPLink are 16 KiB RAM and 128 KiB of Flash with bootloader (or 64 KiB without bootloader). If you disable all features except CMSIS-DAP you might be able to fit into 32 KiB Flash and 6 KiB RAM with a little bit of hacking.

**Comment by H4w4k4** [comment]
Hello,

Thank you for the explanation. 

I have selected the STM32L412K8T6. Might you be able to confirm that it will work as fully featured HIC (without bootloader)?

If I understand correctly the bootloader is only needed for updating the HIC through a drag-n-drop process. That MCU has a USB DFU mode, so updating it shouldn't be an issue without the BL.

(Sorry for the delayed reply, this is a side-project for me)
